airflow/operators/latest_only_operator.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import logging
+
+from airflow.models import BaseOperator, TaskInstance
+from airflow.utils.state import State
+from airflow import settings
+
+
+class LatestOnlyOperator(BaseOperator):
+    """
+    Allows a workflow to skip tasks that are not running during the most
+    recent schedule interval.
+
+    If the task is run outside of the latest schedule interval, all
+    directly downstream tasks will be skipped.
+    """
+
+    ui_color = '#bb6bd6'  # nyanza
+
+    def execute(self, context):
+        start_date = context['ti'].start_date
```

Comment:
Check out https://gist.github.com/r39132/ad9601e5d056f655936c0832ca772d6d

start_date won't work. The problem is that it's fixed and doesn't change with your tasks, especially, if passed via default_args, such as shown in https://gist.github.com/r39132/2e6e8353d42852cdb8f7717a3b1998b9

In other words, it's set one time for most people, at the time the dag is first parsed.

Also, the computation of left and right doesn't seem right. 

For an hourly dag, `context['execution_date']` of noon will run at 1p. The left_window would be noon, the right_window would be 1pm and now would be a few seconds past 1p.